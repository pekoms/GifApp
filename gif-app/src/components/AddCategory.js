import React,{useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')
    const handleInputChange=(e)=>{
        setInputValue(e.target.value);

        //console.log('HandleInputChange Llamado')
    }

    const handleSubmit=(e)=>{
        e.preventDefault();
        //console.log("Submit hecho")
        if(inputValue.trim().length>2){
            setCategories(cats=>[inputValue,...cats,]);
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
        <div>
            <p>{inputValue}</p>
           <input type="text" value={inputValue} onChange={handleInputChange}></input>
        </div>
        </form>
    )

 
}

AddCategory.propTypes={
    setCategories:PropTypes.func.isRequired
}