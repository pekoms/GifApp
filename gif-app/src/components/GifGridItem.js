import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem=({id,title,url})=> {
    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}></img>
            <p>{title}</p>

        </div>
    )
}

GifGridItem.prototype={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,


}