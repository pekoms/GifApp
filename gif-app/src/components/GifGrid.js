import { useEffect } from "react"
import { useState } from "react"
import {React} from 'react'
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";


export default function GifGrid({category}) {

      const {data:images,loading} = useFetchGifs(category);
    
    return (
        <>
        <h3 className="animate__fadeInRightBig">{category}</h3>
        {loading && <p className="animate__flash">Loading</p>}
        <div className="card-grid animate__fadeInLeft">
            
            
            {
                images.map(img=>(
                    <GifGridItem key={img.id}{...img}></GifGridItem>
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired
}