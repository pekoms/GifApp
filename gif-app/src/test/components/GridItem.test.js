import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

const title='Un Título';
const url ='https://localhost/algo.jpg';
const wrapper = shallow(<GifGridItem title={title} url={url} />)

describe('Pruebas en <GridItem></GridItem>',()=>{
    test('Debe mostrar el componente correctamente',()=>{
    
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un párrafo con el title',()=>{
        const p=wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    })

    test('debe tener la imagen igual al url y alt de los props',()=>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('debe de tener animate__bounce',()=>{
        const div = wrapper.find('div');
        const className=div.prop('className');
        expect(className.includes('animate__bounce')).toBe(true);
    })
})
