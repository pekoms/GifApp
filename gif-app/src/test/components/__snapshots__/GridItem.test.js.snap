// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridItem></GridItem> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un Título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Título
  </p>
</div>
`;
